transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptor.java

```diff
@@ -593,6 +610,80 @@ public void removeBridgeFilters(IoFilterChain filterChain) {
         }
     }
 
+    // If HTTP's X-FORWARDED-FOR header is set, then use it's value to determine the remote
+    // IP address. If X-FORWARDED-FOR header is NOT set, then use IoSession to determine
+    // the remote IP address.
+    private String getRemoteIpAddress(IoSession session, HttpRequestMessage httpRequest) {
+        String remoteIpAddr = httpRequest.getHeader(HEADER_X_FORWARDED_FOR);
+
+        if (remoteIpAddr == null) {
+            SocketAddress socketAddr = session.getRemoteAddress();
+
+            if (session instanceof AbstractBridgeSession) {
+                IoSession tcpSession = getTcpSession(session);
+                if (tcpSession == null) {
+                    return null;
+                }
+                socketAddr = tcpSession.getRemoteAddress();
+            }
+
+            InetAddress inetAddr = ((InetSocketAddress)socketAddr).getAddress();
+            remoteIpAddr = inetAddr.getHostAddress();
+
+            if (logger.isDebugEnabled()) {
+                logger.debug("Using IoSession to determine remote IP address: [%s].", remoteIpAddr);
+            }
+        }
+        else {
+            // X-FORWARDED-FOR HTTP header is inserted by proxies to identify the remote IP address.
+            // If the proxies are chained, then X-FORWARDED-FOR header can contain multiple IP
+            // addresses separated by commas. In such a scenario, the first IP address represents
+            // the remote IP address.
+            int index = remoteIpAddr.indexOf(',');
+            if (index > -1) {
+                remoteIpAddr = remoteIpAddr.substring(0, index).trim();
```

Comment:
HttpRequestMessage.getHeader() returns the first value if there are more values in the header. So no need to parse the header value